### PR TITLE
8302155: [AIX] NUM_LCPU is not required variable

### PR DESCRIPTION
**Pull request by offamitkumar**
We don't need NUM_LCPU, it's just a small code change which removes this variable. I've tested it on AIX and cores are detected successfully with changes. 
@backwaterred please take look at it. 
 Suggestions are welcomed :)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302155](https://bugs.openjdk.org/browse/JDK-8302155): [AIX]  NUM_LCPU is not required variable

### Reviewers
 * [Tyler Steele](https://openjdk.org/census#tsteele) (@backwaterred - Committer) ⚠️ Review applies to [08a528f7](https://git.openjdk.org/jdk/pull/12494/files/08a528f7aa3dfffa9ffab375b9d9ad89aae3c368)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12494/head:pull/12494` \
`$ git checkout pull/12494`

Update a local copy of the PR: \
`$ git checkout pull/12494` \
`$ git pull https://git.openjdk.org/jdk pull/12494/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12494`

View PR using the GUI difftool: \
`$ git pr show -t 12494`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12494.diff">https://git.openjdk.org/jdk/pull/12494.diff</a>

</details>

### CODE REVIEW

**Review by backwaterred — APPROVED** [review]
I wanted to point out these changes are semantically different in the error case. If we don't get the output we expect from lparstat we potentially overwrite the NUM_CORES var with the empty string. Previously, the code protected against this by testing if the result stored in the temp variable was non-empty before committing to them.

Checking the [documentation](https://www.ibm.com/docs/en/aix/7.2?topic=l-lparstat-command) for AIX 7.1-7.3, it seems like the syntax is stable, so removing this check seems reasonable to me.

File: make/autoconf/build-performance.m4 [hunk 1]
```diff
@@ -38,10 +38,7 @@ AC_DEFUN([BPERF_CHECK_CORES],
     # Looks like a MacOSX system
     NUM_CORES=`/usr/sbin/sysctl -n hw.ncpu`
   elif test "x$OPENJDK_BUILD_OS" = xaix ; then
-    NUM_LCPU=`lparstat -m 2> /dev/null | $GREP -o "lcpu=[[0-9]]*" | $CUT -d "=" -f 2`
-    if test -n "$NUM_LCPU"; then
-      NUM_CORES=$NUM_LCPU
-    fi
+    NUM_CORES=`lparstat -m 2> /dev/null | $GREP -o "lcpu=[[0-9]]*" | $CUT -d "=" -f 2`
```
Comment (magicus) [inline make/autoconf/build-performance.m4:41]: The assignment of `NUM_CORES=0` on line 29 was to cover the case that `lparstat` failed. If you are certain that it works, fine, make this change. But please also remove the unnecessary initialization of `NUM_CORES` then.

**Review by magicus — CHANGES_REQUESTED** [review]
(github needs me to repeat the comment here)

**Review by magicus — APPROVED** [review]
The merge warning from the bots sounded a bit scary; I don't understand what git move you did there, but the end result seems to look fine anyway.

**Comment by magicus** [comment]
I can sponsor this change, unless you have some other sponsor already.

**Comment by offamitkumar** [comment]
Sure @magicus , 
/integrate

**Pull request closed without merging**